### PR TITLE
perf(orjson): Remove unused options

### PR DESCRIPTION
**Pull request by loewenheim**
The options

* `integrations.slack.enable-orjson`
* `auth.enable-orjson`
* `backup.enable-orjson`
* `event-manager.enable-orjson`
* `eventstore.enable-orjson`
* `flagpole.enable-orjson`

introduced in https://github.com/getsentry/sentry/pull/69409 and https://github.com/getsentry/sentry/pull/69653 are already set to `1.0` on US. This removes them in favor of direct calls to `orjson`.

It also removes some automatic fixtures that were only used to set those options to 0 in tests.

### CODE REVIEW

File: src/sentry/backup/crypto.py [hunk 1]
```diff
@@ -135,9 +135,7 @@ def create_encrypted_export_tarball(json_export: json.JSONData, encryptor: Encry
     pem = encryptor.get_public_key_pem()
     data_encryption_key = Fernet.generate_key()
     backup_encryptor = Fernet(data_encryption_key)
-    encrypted_json_export = backup_encryptor.encrypt(
-        json.dumps_experimental("backup.enable-orjson", json_export).encode()
-    )
+    encrypted_json_export = backup_encryptor.encrypt(json.dumps_orjson(json_export).encode())
```
[the remark this answers is not in the document]
Comment (loewenheim) [reply src/sentry/backup/crypto.py:138]: Hm, fair point. What would you suggest to work around the fact that `orjson` dumps to `bytes`, in contrast to `json`?

[the remark this answers is not in the document]
Comment (asottile-sentry) [reply src/sentry/backup/crypto.py:138]: maybe we should also have an `orjson_dumpsb` ?

File: src/sentry/integrations/slack/webhooks/event.py [hunk 2]
```diff
@@ -156,9 +156,7 @@ def on_link_shared(self, request: Request, slack_request: SlackDMRequest) -> boo
                 return True
 
             # Don't unfurl the same thing multiple times
-            seen_marker = hash(
-                json.dumps_experimental("integrations.slack.enable-orjson", (link_type, args))
-            )
+            seen_marker = hash(json.dumps_orjson((link_type, list(args))))
```
[the remark this answers is not in the document]
Comment (asottile-sentry) [reply src/sentry/integrations/slack/webhooks/event.py:159]: is the data in the test wrong?

File: src/sentry/backup/imports.py [hunk 3]
```diff
@@ -176,7 +177,7 @@ def _import(
     if len(DELETED_MODELS) > 0 or len(DELETED_FIELDS) > 0:
         # Parse the content JSON and remove fields and models that we have marked for deletion in the
         # function.
-        content_as_json = json.loads_experimental("backup.enable-orjson", content)  # type: ignore[arg-type]
+        content_as_json = orjson.loads(content)  # type: ignore[arg-type]
```
Comment (anonrig) [inline src/sentry/backup/imports.py:180]:
```suggestion
        content_as_json = orjson.loads(content)
```

File: src/sentry/backup/imports.py [hunk 4]
```diff
@@ -189,7 +190,7 @@ def _import(
                 del content_as_json[i]
 
         # Return the content to byte form, as that is what the Django deserializer expects.
-        content = json.dumps_experimental("backup.enable-orjson", content_as_json)
+        content = orjson.dumps(content_as_json).decode()
```
Comment (anonrig) [inline src/sentry/backup/imports.py:193]: If your change is correct, than previous comment needs to be changed:

```suggestion
        content = orjson.dumps(content_as_json)
```

File: src/sentry/event_manager.py [hunk 5]
```diff
@@ -2437,13 +2437,11 @@ def _get_severity_score(event: Event) -> tuple[float, str]:
                 response = severity_connection_pool.urlopen(
                     "POST",
                     "/v0/issues/severity-score",
-                    body=json.dumps_experimental("event-manager.enable-orjson", payload),
+                    body=orjson.dumps(payload).decode(),
```
Comment (anonrig) [inline src/sentry/event_manager.py:2440]: Passing bytes to body is faster:

```suggestion
                    body=orjson.dumps(payload),
```

File: src/sentry/event_manager.py [hunk 6]
```diff
@@ -2789,9 +2787,7 @@ def _materialize_event_metrics(jobs: Sequence[Job]) -> None:
         job["event"].data["_metrics"] = event_metrics
 
         # Capture the actual size that goes into node store.
-        event_metrics["bytes.stored.event"] = len(
-            json.dumps_experimental("event-manager.enable-orjson", dict(job["event"].data.items()))
-        )
+        event_metrics["bytes.stored.event"] = len(orjson.dumps(dict(job["event"].data.items())))
```
Comment (anonrig) [inline src/sentry/event_manager.py:2790]: Nit: I'm not sure which one is more correct:

```suggestion
        event_metrics["bytes.stored.event"] = len(orjson.dumps(dict(job["event"].data.items())).decode())
```

File: src/sentry/eventstore/models.py [hunk 7]
```diff
@@ -519,7 +519,7 @@ def get_raw_data(self, for_stream: bool = False) -> Mapping[str, Any]:
 
     @property
     def size(self) -> int:
-        return len(json.dumps_experimental("eventstore.enable-orjson", dict(self.data)))
+        return len(orjson.dumps(dict(self.data)).decode())
```
Comment (anonrig) [inline src/sentry/eventstore/models.py:522]: Can we get rid of `dict()` call? It makes an unnecessary copy

```suggestion
        return len(orjson.dumps(self.data).decode())
```

File: src/sentry/integrations/slack/webhooks/action.py [hunk 8]
```diff
@@ -468,19 +464,19 @@ def open_archive_dialog(self, slack_request: SlackActionRequest, group: Group) -
...
             payload = {
-                "view": json.dumps_experimental("integrations.slack.enable-orjson", modal_payload),
+                "view": orjson.dumps(modal_payload).decode(),
                 "trigger_id": slack_request.data["trigger_id"],
             }
             headers = {"content-type": "application/json; charset=utf-8"}
             slack_client.post(
                 "/views.open",
-                data=json.dumps_experimental("integrations.slack.enable-orjson", payload),
+                data=orjson.dumps(payload).decode(),
```
Comment (anonrig) [inline src/sentry/integrations/slack/webhooks/action.py:479]: Afaict this is better:

```suggestion
                data=orjson.dumps(payload),
```

**Review by anonrig — APPROVED** [review]

**Pull request merged**